wezterm-gui/src/main.rs

```diff
@@ -508,7 +517,22 @@ fn run() -> anyhow::Result<()> {
             SetStdHandle(STD_ERROR_HANDLE, stderr.into_raw_file_descriptor());
             */
 
-            std::env::set_current_dir(config::HOME_DIR.as_path())?;
+            // Only change the current directory to the user's home path if
```

Comment:
This doesn't appear to be related to the issue that you set out to resolve in the PR.
Please remove it from here and track this separately, along with an explanation of why it is needed.

FWIW, I don't think this is necessary: this is controlling the current directory of the GUI process itself, and not its child processes.

Reply:
This is for starting wezterm in the current working directory from a GUI context. My use case is starting a terminal from my file manager, which starts the process in the currently opened directory. Since there's no console available, the current directory gets reset to the user's home directory. I'll put this into a separate PR.